pkg/scheduler/objects/application.go

```diff
@@ -1193,6 +1202,19 @@ func (sa *Application) UnSetQueue() {
 	sa.Lock()
 	defer sa.Unlock()
 	sa.queue = nil
+	now := time.Now()
+	sa.finishedTime = &now
+}
+
+// FinishedTimeInUnixNano return the finished time if this application is terminated. Otherwise, it returns null
+func (sa *Application) FinishedTimeInUnixNano() *int64 {
```

Comment:
We might want to support this for other time values also.
Can we generalise this into a utility function? Bases on a time object and use time.IsZero():
```
func ZeroTimeInUnixNano(t time.Time) *int64 {
	if t.IsZero() {
		return nil
	}
	tInt := t.UnixNano()
	return &tInt
}
```

Reply:
the conversion into the `*int64` can then be made as part of the DAO creation